db/schema.go

```diff
@@ -1,45 +1,51 @@
 package db
 
-import (
-	"log"
+type DatabaseLayer int
+
+const (
+	DATABASE_LAYER_EMPTY = iota
+	DATABASE_LAYER_1     // original schema, as of 2019
+	DATABASE_LAYER_2     // hash_pre_enc removed, hash_post_enc renamed to final_hash, encryption_key renamed to padding_key, encryption_key added to blob_entries
 )
 
 func initialSetup() {
-	err := createTables()
-	if err != nil {
-		panic(err)
+	switch determineDatabaseLayer() {
+	case DATABASE_LAYER_EMPTY:
+		err := schemaVersionOne()
```

Comment:
I see no upside to that

Reply:
i want to be able to read the entire current schema without the layer system obfuscating it

Reply:
ok.

I think I'll have the current schema in a comment or something then, or a different file

Reply:
just put it in code and use it to initialize new databases like it already is

Reply:
no i dont like that

Reply:
what’s the point of having the current schema written down and always going through the layers?

Reply:
because obviously i still need the code to go from layer one to layer two, because gb is going to remain backwards compatible. therefore, i could have two separately maintained bits of code to go from 0 to N and from N-1 to N, sure. but the only point of the 0 to N code is that it would be more readable to a human. so i'll just put that as a separate `schema.sql` file that is never executed and is just for human reference.